hyprlang has no other consumers. Keep it private for now.

--- CMakeLists.txt.orig	2024-01-02 21:22:34 UTC
+++ CMakeLists.txt
@@ -41,11 +41,12 @@ find_package(PkgConfig REQUIRED)
 find_package(Threads REQUIRED)
 
 find_package(PkgConfig REQUIRED)
-pkg_check_modules(deps REQUIRED IMPORTED_TARGET wayland-client wayland-protocols cairo pango pangocairo libjpeg libwebp hyprlang>=0.2.0)
+pkg_check_modules(deps REQUIRED IMPORTED_TARGET wayland-client wayland-protocols cairo pango pangocairo libjpeg libwebp)
 
 file(GLOB_RECURSE SRCFILES "src/*.cpp")
 
 add_executable(hyprpaper ${SRCFILES})
+add_subdirectory(subprojects/hyprlang EXCLUDE_FROM_ALL)
 
 target_compile_definitions(hyprpaper PRIVATE "-DGIT_COMMIT_HASH=\"${GIT_COMMIT_HASH}\"")
 target_compile_definitions(hyprpaper PRIVATE "-DGIT_BRANCH=\"${GIT_BRANCH}\"")
@@ -59,6 +60,7 @@ target_link_libraries(hyprpaper PkgConfig::deps)
 include(CPack)
 
 target_link_libraries(hyprpaper PkgConfig::deps)
+target_link_libraries(hyprpaper hyprlang)
 
 target_link_libraries(hyprpaper
         OpenGL
--- subprojects/hyprlang/CMakeLists.txt.orig	2024-01-02 22:10:42 UTC
+++ subprojects/hyprlang/CMakeLists.txt
@@ -17,7 +17,7 @@ file(GLOB_RECURSE SRCFILES CONFIGURE_DEPENDS "src/*.cp
 
 file(GLOB_RECURSE SRCFILES CONFIGURE_DEPENDS "src/*.cpp" "include/hyprlang.hpp")
 
-add_library(hyprlang SHARED ${SRCFILES})
+add_library(hyprlang STATIC ${SRCFILES})
 target_include_directories( hyprlang
     PUBLIC "./include"
     PRIVATE "./src"
